src/openms/source/CHEMISTRY/Residue.cpp

```diff
@@ -568,9 +569,18 @@ namespace OpenMS
 
   void Residue::setModification(const String& name)
   {
-    ModificationsDB* mod_db = ModificationsDB::getInstance();
-    const ResidueModification& mod = mod_db->getModification(name, one_letter_code_, ResidueModification::ANYWHERE);
-    setModification_(mod);
+    try
```

Comment:
I removed the ones I introduced in commit d2ab52f ,
tried to avoid them by using "ModsDB.has()", but that is unfortunately quite useless in these cases.
It uses "findModificationIndex" and returns an ElementNotFound exception, when there is more than one modification with that name. Since the ModsDB contains the same modification multiple times for different residues it can attach to, this happens quite often.
So even if we use "ModsDB.has()" we would still need try catch logic to catch this ElementNotFound exception.

For my tool it is not that important, since all information about cross-link mass and position is always stored separately from the AASequence and can be retrieved, but I always get some warnings, that my cross-links can not be found in ModsDB, when reading in mzid files.